server/src/machines/extruder1/act.rs

```diff
@@ -0,0 +1,74 @@
+use super::{ExtruderV2, ExtruderV2Mode};
+use control_core::{actors::Actor, converters::motor_converter::MotorConverter};
+use std::time::{Duration, Instant};
+
+impl Actor for ExtruderV2 {
+    fn act(
+        &mut self,
+        now_ts: Instant,
+    ) -> std::pin::Pin<Box<dyn std::future::Future<Output = ()> + Send + '_>> {
+        Box::pin(async move {
+            self.inverter.act(now_ts).await;
+            self.pressure_sensor.act(now_ts).await;
+            self.temperature_controller_back
+                .update_heating(now_ts)
+                .await;
+            self.temperature_controller_nozzle
+                .update_heating(now_ts)
+                .await;
+            self.temperature_controller_front
+                .update_heating(now_ts)
+                .await;
+            self.temperature_controller_middle
+                .update_heating(now_ts)
+                .await;
+            self.set_bar();
+
+            if self.mode == ExtruderV2Mode::Standby {
+                self.turn_heating_off();
+            } else if self.mode == ExtruderV2Mode::Heat || self.mode == ExtruderV2Mode::Extrude {
+                self.temperature_controller_front.update_pid(now_ts);
+                self.temperature_controller_middle.update_pid(now_ts);
+                self.temperature_controller_back.update_pid(now_ts);
+                self.temperature_controller_nozzle.update_pid(now_ts);
+            }
```

Comment:
Please put his code into it's own function somewhere else like `update_heating` or something.